src/engine/enginebuffer.cpp

```diff
@@ -676,6 +684,26 @@ void EngineBuffer::slotControlEnd(double v)
     }
 }
 
+void EngineBuffer::slotControlPlaySync(double v)
+{
+    if (v > 0.0) {
+        // Momentarily enable/disable sync, unless sync mode is already
+        // on or has been requested.
+        if (m_pSyncControl->getSyncMode() == SYNC_NONE) {
+            // Set the request to enable/disable, but if the existing request
+            // was already an enable request, restore that original request.
+            // This should prevent a race condition from causing an existing
+            // sync-enable request from being overwritten.
+            const int existing_request = m_iEnableSyncQueued.fetchAndStoreAcquire(
```

Comment:
I think you have to secure the code in requestEnableSync() as well. It will be hard to do all with m_iEnableSyncQueued. You probably need a second variable.